src/keeper.ts

```diff
@@ -231,7 +232,7 @@ class Keeper {
     }
   }
 
-  async processNewBlock(blockNumber: string) {
+  async processNewBlock(blockNumber: number) {
```

Comment:
I think this is implemented to process a single block instead of all the blocks since the last one processed because of ` fromBlock: blockNumber, toBlock: blockNumber }`. It's also possible that this assumption is baked in into other parts of the code - worth to double check.

Reply:
This method is called by the `startProcessNewBlockConsumer` in a while(true) loop. The only change here is that we had blockNumber incorrectly typed as a `string`

Reply:
Thanks for catching. I went in thinking it was ok to skip blocks. We still want to process all block just not "on" every block.